lib/matplotlib/tests/test_datetime.py

```diff
@@ -0,0 +1,394 @@
+        ax.fill_between(...)
+
+    @pytest.mark.xfail(reason="Test for fill_betweenx not written yet")
+    @mpl.style.context("default")
+    def test_fill_betweenx(self):
+        fig, ax = plt.subplots()
+        ax.fill_betweenx(...)
+
+    @pytest.mark.xfail(reason="Test for hexbin not written yet")
+    @mpl.style.context("default")
+    def test_hexbin(self):
+        fig, ax = plt.subplots()
+        ax.hexbin(...)
+
+    @pytest.mark.xfail(reason="Test for hist not written yet")
+    @mpl.style.context("default")
+    def test_hist(self):
+        fig, ax = plt.subplots()
+        ax.hist(...)
+
+    @pytest.mark.xfail(reason="Test for hist2d not written yet")
+    @mpl.style.context("default")
+    def test_hist2d(self):
+        fig, ax = plt.subplots()
+        ax.hist2d(...)
+
+    @pytest.mark.xfail(reason="Test for hlines not written yet")
+    @mpl.style.context("default")
+    def test_hlines(self):
+        fig, ax = plt.subplots()
+        ax.hlines(...)
+
+    @pytest.mark.xfail(reason="Test for imshow not written yet")
+    @mpl.style.context("default")
+    def test_imshow(self):
+        fig, ax = plt.subplots()
+        ax.imshow(...)
+
+    @pytest.mark.xfail(reason="Test for loglog not written yet")
+    @mpl.style.context("default")
+    def test_loglog(self):
+        fig, ax = plt.subplots()
+        ax.loglog(...)
+
+    @pytest.mark.xfail(reason="Test for magnitude_spectrum not written yet")
+    @mpl.style.context("default")
+    def test_magnitude_spectrum(self):
+        fig, ax = plt.subplots()
+        ax.magnitude_spectrum(...)
+
+    @pytest.mark.xfail(reason="Test for matshow not written yet")
+    @mpl.style.context("default")
+    def test_matshow(self):
+        fig, ax = plt.subplots()
+        ax.matshow(...)
+
+    @pytest.mark.xfail(reason="Test for pie not written yet")
+    @mpl.style.context("default")
+    def test_pie(self):
+        fig, ax = plt.subplots()
+        ax.pcolor(...)
+
+    @pytest.mark.xfail(reason="Test for pcolor not written yet")
+    @mpl.style.context("default")
+    def test_pcolor(self):
+        fig, ax = plt.subplots()
+        ax.pcolor(...)
+
+    @pytest.mark.xfail(reason="Test for pcolorfast not written yet")
+    @mpl.style.context("default")
+    def test_pcolorfast(self):
+        fig, ax = plt.subplots()
+        ax.pcolorfast(...)
+
+    @pytest.mark.xfail(reason="Test for pcolormesh not written yet")
+    @mpl.style.context("default")
+    def test_pcolormesh(self):
+        fig, ax = plt.subplots()
+        ax.pcolormesh(...)
+
+    @pytest.mark.xfail(reason="Test for phase_spectrum not written yet")
+    @mpl.style.context("default")
+    def test_phase_spectrum(self):
+        fig, ax = plt.subplots()
+        ax.phase_spectrum(...)
+
+    @mpl.style.context("default")
+    def test_plot(self):
+        N = 6
+        fig, ax = plt.subplots()
+        x = np.array([datetime.datetime(2023, 1, n) for n in range(1, N)])
+        ax.plot(x, range(1, N))
```

Comment:
Should you also test `ax.plot(range(1, N), x)` and/or `ax.plot(x, x)`? It might be good to give an example of what to do when multiple arguments accept units.